plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/NioHttpRequestCreator.java

```diff
@@ -25,11 +25,11 @@ protected void decode(ChannelHandlerContext ctx, FullHttpRequest msg, List<Objec
         if (msg.decoderResult().isFailure()) {
             final Throwable cause = msg.decoderResult().cause();
             final Exception nonError;
-            if (cause instanceof Error) {
+            if (cause instanceof Exception exception) {
```

Comment:
I would like to draw the reviewers attention to this because it is not a straightforward case. I see two cases:

Case 1: The cause can only be an `Error` or an `Exception`, then this refactoring should be ok. 
Case 2: The premise of case 1 is not valid, then the semantics of the code have changed because anything else will be dealt with like an error instead of getting a `ClassCastException`.

I made the assumption that case 1 is more probable because I would expect that this would be coded differently since a `ClassCastException` is an undesirable exception. I also prefer to cast the `Exception` after we specifically check if it the type of the cause with `instanceof`.

What are your thoughts in this matter? Am I missing anything?

Reply:
Reviewed and approved.